src/sentry/web/frontend/accounts_twofactor.py

```diff
@@ -87,12 +112,13 @@ def remove(self, request, interface):
             form = ConfirmPasswordForm(request.user, request.POST)
             if 'password' in form.fields:
                 if form.is_valid():
-                    self.delete_authenticator(interface)
-                    return HttpResponseRedirect(reverse('sentry-account-settings-2fa'))
-                else:
-                    form.errors['__all__'] = ['Invalid password.']
+                    if request.user.check_password(form.cleaned_data['password']):
```

Comment:
This check is redundant. The form wouldn't be validated if the password was wrong. This happens in English `clean_password` method.